narwhals/_arrow/series.py

```diff
@@ -997,9 +966,64 @@ def rolling_mean(
             )
             / count_in_window
         )
-        if center:
-            result = result[offset_left + offset_right :]
-        return result
+        return result[offset:]
+
+    def rolling_var(
+        self: Self,
+        window_size: int,
+        *,
+        min_periods: int | None,
+        center: bool,
+        ddof: int,
+    ) -> Self:
+        import pyarrow.compute as pc  # ignore-banned-import
+
+        min_periods = min_periods if min_periods is not None else window_size
+        padded_series, offset = pad_series(self, window_size=window_size, center=center)
+
+        cum_sum = padded_series.cum_sum(reverse=False).fill_null(strategy="forward")
+        rolling_sum = (
+            cum_sum - cum_sum.shift(window_size).fill_null(0)
+            if window_size != 0
+            else cum_sum
+        )
+
+        cum_sum_sq = (
+            padded_series.__pow__(2).cum_sum(reverse=False).fill_null(strategy="forward")
+        )
+        rolling_sum_sq = (
+            cum_sum_sq - cum_sum_sq.shift(window_size).fill_null(0)
+            if window_size != 0
+            else cum_sum_sq
+        )
+
+        valid_count = padded_series.cum_count(reverse=False)
+        count_in_window = valid_count - valid_count.shift(window_size).fill_null(0)
+
+        result: Self = self._from_native_series(
```

Comment:
again, mypy doesn't really know what happens after the division, so we annotate this, otherwise it yells with:

>  Returning Any from function declared to return "Self"  [no-any-return]